Eastern-Research-Group/mywaterway
# Feature/163 add streamgage sparklines

## Related Issues:
* https://jira.epa.gov/browse/HMW-163

## Main Changes:
* Add sparkline chart showing averaged daily data for the past week to USGS streamgages for parameters that have data in the USGS daily values web service.
* Begins moving fetched data into a dedicated context component, to break up the "locationSearch" context component of doom.

## Steps To Test:
1. Search any HUC (e.g. http://localhost:3000/community/020700100102/overview). From the Community/Overview tab, toggle "Current Water Conditions" switch in the "Monitoring & Sensors" subtab and verify a sparkline chart is displayed in the layerlist to the left of a measurement (**NOTE**: not all parameters have weekly data – I've found _most_ do not).
2. The USGS streamgages displaying at all _should_ be enough to ensure nothing broke in moving the fetched data out of the `locationSearch` context component into the new `FetchedData` context component, but it's possible something could've been missed with the resetting of data.

## TODO:
- [ ] Fix tooltip bug – it's value currently doesn't update every time as you hover over the sparkline. I want to show EPA this first to get sign-off on overall effect first, before digging into the tooltip not updating (could be a debouncing issue?)
